### PR TITLE
Added instance method tofile(), tested it

### PR DESCRIPTION
**Pull request by cressidasuphina**

### CODE REVIEW

**Review by Fayad-Alman — CHANGES_REQUESTED** [review]
@cressidasuphina I believe that the tofile function has not yet been implemented in ivy/data_classes/array/general.py, which would explain why the tests on this function are failing
Please add the appropriate function in the general.py class and request another review when you're finished
Many thanks for your contribution!

**Comment by cressidasuphina** [comment]
Hi @Fayad-Alman thank you for your comment, I did what you asked. Should i add function to_file() to ivy\functional\ivy\general.py too?

File: ivy/data_classes/array/general.py [hunk 1]
```diff
@@ -588,6 +588,42 @@ def to_list(self: ivy.Array, /) -> List:
...
+
+        >>> x = ivy.array([1, 2, 3])
+        >>> x.to_file('data.txt', sep=',', format_='%d')
+
+        Notes
+        -----
+        The data produced by this method can be recovered using appropriate methods or functions depending on the data type.
+        """
+
+        return ivy.to_file(self)
```
Comment (Fayad-Alman) [inline ivy/data_classes/array/general.py:624]: @cressidasuphina You need to add the correct parameters for this return statement. ivy.to_file takes in self, fid, sep, and format. Please add these to the return statement and request another review

**Review by Fayad-Alman — CHANGES_REQUESTED** [review]

File: ivy/functional/frontends/numpy/ndarray/ndarray.py [hunk 2]
```diff
@@ -315,6 +315,9 @@ def prod(
             out=out,
         )
 
+    def tofile(self, fid, sep="", format="%s"):
+        return self._ivy_array.to_file(fid, sep=sep, format=format)
```
Comment (Fayad-Alman) [inline ivy/functional/frontends/numpy/ndarray/ndarray.py:319]: @cressidasuphina A small error here; the parameter should be format_ , not format (if you check the to_file() method in array/general.py you will see why)

**Review by Fayad-Alman — CHANGES_REQUESTED** [review]

File: ivy/functional/frontends/numpy/ndarray/ndarray.py (hunk 2 above)
Comment (cressidasuphina) [reply ivy/functional/frontends/numpy/ndarray/ndarray.py]: @Fayad-Alman I'm sorry, I haven't noticed.

**Review by Fayad-Alman — APPROVED** [review]
LGTM! Thanks for your contributions!

**Pull request merged**